### PR TITLE
Add course averages

### PR DESCRIPTION
**Pull request by SamZhang02**
This PR adds the course averages using McGill Enhanced's crowdsourced sheet. The use was okay'ed by Koz and credits are given within the Class Average component.

The data is accessed using the Google Sheets API and a python script in `/tools`. It gets written as a Json in `/assets` and will be periodically updated (I am thinking every month or two). I chose to have to as a static asset instead of seeding it in the db because its a relative light file and it doesn't need to be updated by anyone else than the maintainers. This choice consistent with McGill Enhanced.

I'm still not sure if I can make the UI better tho, here is what it looks like right now:

CleanShot 2024-05-02 at 12 35 03

### CODE REVIEW

**Comment by terror** [comment]
Nice work!

**Comment by 39bytes** [comment]
What about folding the averages to max 5-6 most recent then having a "show all" button to expand it? It feels like it can get a bit long if you show all of them by default
image

(also, thoughts on making the font weight on the grades slightly more bold? i gave it `font-medium` in this screenshot)

**Comment by 39bytes** [comment]
Another useful thing I just thought of, it might be cool to put the instructor that taught during that term for each course average, can't think of a good way to fit it into the UI at the moment though, maybe just some really small text under the course term in the list?

File: client/src/components/CourseAverages.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    >
+      <h2 className='mb-2 mt-1 text-xl font-bold leading-none text-gray-700 dark:text-gray-200'>
+        Class Averages
+      </h2>
+
+      <hr className='my-2 mt-5 w-full border border-neutral-200 dark:border-neutral-700' />
+
+      {averages
+        .sort((a, b) => compareTerms(a.term, b.term))
+        .reverse()
```
[the remark this answers is not in the document]
Comment (SamZhang02) [reply client/src/components/CourseAverages.tsx:23]: shit you're right

**Comment by SamZhang02** [comment]
> What about folding the averages to max 5-6 most recent then having a "show all" button to expand it? It feels like it can get a bit long if you show all of them by default. image
> 
> (also, thoughts on making the font weight on the grades slightly more bold? i gave it `font-medium` in this screenshot)

yesss good idea i'll hop on this

> Another useful thing I just thought of, it might be cool to put the instructor that taught during that term for each course average, can't think of a good way to fit it into the UI at the moment though, maybe just some really small text under the course term in the list?

That's a really great idea yeah, I too struggle to think of the UI for now

**Comment by 39bytes** [comment]
Also one more thing, the text should scale better on mobile (too big)
image

**Comment by SamZhang02** [comment]
I addressed most comments except the one except the one about showing the prof of each semester, we may need some brainstorming for that hmmm

Maybe its better in a different PR? up to you guys

**Comment by 39bytes** [comment]
> I addressed most comments except the one except the one about showing the prof of each semester, we may need some brainstorming for that hmmm
> 
> Maybe its better in a different PR with some time for brainstorm? up to you guys

I just added it, what do you think?
image

**Review by 39bytes — APPROVED** [review]

**Pull request merged**